realhf/impl/model/nn/real_llm_parallel.py

```diff
@@ -35,90 +37,116 @@
     ROW_LINEAR_KEYS = [".attn.c_proj", ".mlp.fc2_weight"]
 
 
-def tensor_slice_partition_fn(
-    tensor: torch.Tensor,
-    mp_rank: Optional[int],
-    mp_world_size: int,
-    dim: Optional[int],
-) -> Union[List[torch.Tensor], torch.Tensor]:
-    """Partition a pytorch tensor for model parallelism."""
-    if dim is None:
-        splits = [tensor for _ in range(mp_world_size)]
-    else:
-        assert tensor.shape[dim] % mp_world_size == 0
-        splits = torch.split(tensor, tensor.shape[dim] // mp_world_size, dim=dim)
-    if mp_rank is None:
-        return [s.contiguous() for s in splits]
-    else:
-        return splits[mp_rank].contiguous()
+class PartitionFunctor:
+    """An abstract class for partitioning tensors, tensor shapes, or intervals
+    in a flattened view.
 
+    See concrete implementations below.
+    """
```

Comment:
Why use "functor" classes instead of functions? BTW I am not sure about the definition of functor and do not really understand why these partitioning functions are related to this term.

Reply:
I don't think this change is necessary either. Reverted.